sdk/identity/azure-identity/src/main/java/com/azure/identity/AzurePipelinesCredential.java

```diff
@@ -75,13 +75,23 @@ public class AzurePipelinesCredential implements TokenCredential {
                     HttpRequest request = new HttpRequest(HttpMethod.POST, url);
                     request.setHeader(HttpHeaderName.AUTHORIZATION, "Bearer " + systemAccessToken);
                     request.setHeader(HttpHeaderName.CONTENT_TYPE, "application/json");
+                    request.setHeader(HttpHeaderName.fromString("X-TFS-FedAuthRedirect"), "Suppress");
                     try (HttpResponse response = httpPipeline.sendSync(request, Context.NONE)) {
                         String responseBody = response.getBodyAsBinaryData().toString();
                         if (response.getStatusCode() != 200) {
-                            throw LOGGER.logExceptionAsError(new ClientAuthenticationException("Failed to get the client assertion token "
+                            String xVssHeader = response.getHeaderValue(HttpHeaderName.fromString("x-vss-e2eid"));
+                            String xMsEdgeRefHeader = response.getHeaderValue(HttpHeaderName.fromString("x-msedge-ref"));
+                            String message = "Failed to get the client assertion token "
                                 + responseBody
-                                + System.lineSeparator()
-                                + "For troubleshooting information see https://aka.ms/azsdk/java/identity/azurepipelinescredential/troubleshoot.", response));
+                                + System.lineSeparator();
```

Comment:
Don't use `System.lineSeparator` when possible if the value may be logged, these are stripped out of logs and may result in a poor reading experience if logs need to be reviewed. In this case I'd recommend the classic comma-delimited pattern (`Failed to get the client assertion token <response body>. x-vss-e2eid: <header val>, x-msedge-ref: <header val>. ...`)